service/controller/resource/s3object/desired.go

```diff
@@ -139,3 +129,13 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 
 	return output, nil
 }
+
+// hashBucketObject returns the given object with object.Hash set to the hash of object.Body.
+func hashBucketObject(object BucketObjectState) BucketObjectState {
```

Comment:
Using `sha512` here so that the hash can be used for ignition verification hash in the bootstrap ignition as well.